src/web/components/Lazy.tsx

```diff
@@ -0,0 +1,40 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { useHasBeenSeen } from '@root/src/web/lib/useHasBeenSeen';
+
+type Props = {
+    children: JSX.Element;
+    margin: number;
+};
+
+// Ensure the ref wrapper expands. This is used for componenents like
+// MostViewedRightWrapper that needs to check it's parent's height
+const flexGrowStyles = css`
+    display: flex;
+    flex-grow: 1;
+`;
+
+export const Lazy = ({ children, margin }: Props) => {
+    const [hasBeenSeen, setRef] = useHasBeenSeen({
+        rootMargin: `${margin}px`,
+    });
+
+    if (typeof setRef !== 'function') {
+        return null;
+    }
+
+    // Why do we check to see if we're disabled here? Because we
+    // use this as a flag to know when a component is
+    // being loaded as part of a Chromatic story or not so that
+    // we can prevent lazy loading our storybook snapshots that we
+    // use for visual regression
+    const renderChildren = hasBeenSeen || Lazy.disabled;
```

Comment:
`Lazy.disabled` iv never seen this pattern of adding booleans to a const in JS, really interesting

Reply:
Me neither. It feels a little bit like cheating in that you're breaking the react contract and not using props but in this case `.storybook/config` isn't part of the react context so props aren't an option.

Reply:
I think if done sparingly it can be a handy trick, i think it makes sense here 👍